istioctl/pkg/convert/ingress.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2018 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package convert
+
+import (
+	"k8s.io/api/extensions/v1beta1"
+
+	"istio.io/istio/pilot/pkg/config/kube/ingress"
+	"istio.io/istio/pilot/pkg/model"
+)
+
+// IstioIngresses converts K8s extensions/v1beta1 Ingresses with Istio rules to v1alpha3 gateway and virtual service
+func IstioIngresses(ingresses []*v1beta1.Ingress, domainSuffix string) ([]model.Config, error) {
+	
+	out := make([]model.Config, 0)
+
+	for _, ingrezz := range ingresses {
+		gateway, virtualService := ingress.ConvertIngressV1alpha3(*ingrezz, domainSuffix)
+		out = append(out, gateway)
+		out = append(out, virtualService)
+	}
```

Comment:
This logic is wrong. You need only one gateway for all of istio ingress with wildcard host *, and ports 80/443. And then only one virtual service pinned to the gateway. 

What you need to do here is 
For each ingress, get the gateway.server(443) and append it to the Uber gateway.servers if and only if server has non wildcard host (ie not *). If it has only wildcard, append only once (cant have multiple tls servers on wildcard host).

For virtual services, append all non empty match blocks to the Uber virtual service (where non empty is a match that has authority or path match - not prefix /).

Reply:
I don't think we want this behavior long term (just one Gateway, I mean). I'm not 100% sure I'll need to change it to get all of the ingress tests passing, but in general we should support multiple 80/443 Gateways for ingress. I don't see a compelling reason we can't just blindly create a duplicate gateway per ingress.